File: recipes/concatenate_string_literal/java/ConcatenateStringLiteral.java
public class ConcatenateStringLiteral {
  public static void main(String[] args) {
    // Concatenate string literal simply
    System.out.println("Panda " + "is " + "cute.");

    // Concatenate string literal with delimiter
    // Stinrg.join() is added since java8
    System.out.println(String.join(", ", "Panda", "Giant panda", "Lesser panda"));

    // As needed, please use StringBuilder
    StringBuilder sb = new StringBuilder("Panda");
    sb.append(" is great!");
    System.out.println(sb);
  }
}
                           

File: recipes/concatenate_string_literal/cs/ConcatenateStringLiteral.cs
using System;
class ConcatenateStringLiteral {
    static void Main() {
        // Concatenate string literal simply using +
        Console.WriteLine("Panda " + "is " + "cute.");

        // We can use also +=
        string text = "Panda ";
        text += "is ";
        text += "cute.";
        Console.WriteLine(text);

        // We can use also Concat() of String
        text = null;
        text = String.Concat("Panda ", "is ");
        text = String.Concat(text, "cute.");
        Console.WriteLine(text);

        // If performance is important, we should use Append() of StringBuilder
        System.Text.StringBuilder sb = new System.Text.StringBuilder();
        sb.Append("Panda ");
        sb.Append("is ");
        sb.Append("cute.");
        Console.WriteLine(sb.ToString());
    }
}
        
